components/ecu/ecu_firmware/Core/Src/can.c

```diff
@@ -7,18 +7,25 @@
  */
 
 #include "can.h"
+#include "main.h"
 
 CAN_TxHeaderTypeDef current_message_header = {
-    0x624, 0, CAN_ID_STD, CAN_RTR_DATA, 6, DISABLE
+    0x450, 0, CAN_ID_STD, CAN_RTR_DATA, 3, DISABLE
 }; 
 
 void CAN_send_current(int currentVal) {
+    int doc_coc = 0;
     uint32_t begin_tick = HAL_GetTick();
     uint32_t * pTxMailbox = 0;
     HAL_StatusTypeDef status;
     currentVal /= 100;
-    uint8_t data[] = {(uint8_t)currentVal, (uint8_t)(currentVal >> 8), 0, 0, 0, 0};
+
+    if (HAL_GPIO_ReadPin(OC_LATCH_SET_GPIO_Port, OC_LATCH_SET_Pin) == 1) doc_coc = 0;
+    else doc_coc = 1;
+
+    uint8_t data[] = {(uint8_t)currentVal, (uint8_t)(currentVal >> 8), 0, (uint8_t) doc_coc, 0};
```

Comment:
First off, note that as of right now Mischa (@mjohal67) has changed the format of this message in the CAN table to this:

![image](https://user-images.githubusercontent.com/55331709/221384724-5ef163c1-941f-4916-af9b-2273be20384e.png)

With Mischa's change as of now, DOC_COC is now bit 8, and the current reading has a range of -75A to 75A, placed in bits 0 through 7 (ie. the first byte).

Note that we always number bits starting from 0 ("zero-indexing").

What you have here places `doc_coc` in bit number 16, not 8 (or 9, as previously specified). Remember that each item in this `uint8_t` array is 8 bits. Don't make this change just yet though. I think DOC_COC is actually better off in bit 16 and we should change the format specification; this leaves room to place the LV current field in bits 8 through 15 (the entire 2nd byte).

A change I would like you to make here right now is to update the placement and scaling of the current value (`current_val`) according to Mischa's specification in the table. At the moment it is 16 bits (first 2 bytes), please change it to just one byte as in the table. Also please change the scale factor; note that the input to the function is a signed integer in 100ths of an Amp.

Finally, I'd like you to add a function description comment to this function. A function description comment looks like this:

```c
/**
 * @brief One-line description of what the function does
 *
 * More details if necessary...
 *
 * @param parameter_name Parameter description
 * @param another_parameter_name Another parameter description
 * @returns description of what the function returns, if there is a return value
 */
```

Reply:
After Andrews comment on byte-alignment, I've adjusted the CAN protocol - you can find the latest version on the document, and in the photo below. 
![image](https://user-images.githubusercontent.com/61034360/221387344-b23ea60b-5cb0-42f2-a5c9-62a86111d050.png)